app/src/page/discover/view.js

```diff
@@ -41,9 +41,59 @@ class DiscoverPage extends React.PureComponent {
       }
     });
 
-    const { fetchFeaturedUris, fetchRewardedContent } = this.props;
+    const {
+      fetchFeaturedUris,
+      fetchRewardedContent,
+      fetchSubscriptions
+    } = this.props;
+
     fetchFeaturedUris();
     fetchRewardedContent();
+    fetchSubscriptions();
+  }
+
+  subscriptionForUri = (uri, channelName) => {
+    const { allSubscriptions } = this.props;
+    const { claimId, claimName } = parseURI(uri);
+
+    if (allSubscriptions) {
+      for (let i = 0; i < allSubscriptions.length; i++) {
+        const sub = allSubscriptions[i];
+        if (sub.claim_id === claimId && sub.name === claimName && sub.channel_name === channelName) {
+          return sub;
+        }
+      }
+    }
+
+    return null;
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { allSubscriptions, unreadSubscriptions, enabledChannelNotifications } = this.props;
+
+    const utility = NativeModules.UtilityModule;
+    if (utility) {
+      if (prevProps.unreadSubscriptions &&
```

Comment:
This format isn't the most legible, can you split out this argument as a separate line?